back/app/models/user.rb

```diff
@@ -293,7 +293,23 @@ def invite_not_pending?
   end
 
   def full_name
-    [first_name, last_name].compact.join(' ')
+    return [first_name, last_name].compact.join(' ') unless no_name?
+
+    [anon_first_name, anon_last_name].compact.join(' ')
+  end
+
+  def no_name?
```

Comment:
If the user specifies a last name, but no first name, what is the desired behaviour? We the user be shown as e.g. "User Smith"?

Reply:
Very small remark: I would have a slight preference to call the predicate `name?` instead of `no_name?` and negate the logic in the body so that we can write `name?` instead of `!no_name?` or `unless no_name?`, but it's really fine if you want to keep it like this.